### PR TITLE
[SYCLomatic] Remove the onedpl dependency in dnnl_utils.hpp

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Review by ziranzha — APPROVED** [review]

**Review by intwanghao — APPROVED** [review]
LGTM

File: clang/test/dpct/dnn/reduction.cu [hunk 1]
```diff
@@ -2,11 +2,13 @@
 // RUN: FileCheck --input-file %T/reduction/reduction.dp.cpp --match-full-lines %s
 // RUN: %if build_lit %{icpx -c -fsycl %T/reduction/reduction.dp.cpp -o %T/reduction/reduction.dp.o %}
 
-// CHECK: #include <dpct/dnnl_utils.hpp>
-// CHECK: #include <sycl/sycl.hpp>
-// CHECK: #include <dpct/dpct.hpp>
-// CHECK: #include <iostream>
-// CHECK: #include <vector>
+// CHECK: #include <oneapi/dpl/execution>
```
Comment (zhimingwang36) [inline clang/test/dpct/dnn/reduction.cu]: why here need more header files, dpl/algorithm?

Comment (zhiweij1) [reply clang/test/dpct/dnn/reduction.cu]: Fixed

**Review by intwanghao — APPROVED** [review]
LGTM

**Review by abhilash1910 — COMMENTED** [review]
LGTM !

**Review by ziranzha — APPROVED** [review]

**Pull request merged**